firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+TabToolbarDelegate.swift

```diff
@@ -36,6 +36,59 @@ extension BrowserViewController: TabToolbarDelegate, PhotonActionSheetProtocol {
         UIAccessibility.post(notification: .layoutChanged, argument: dataClearanceContextHintVC)
     }
 
+    // Starts a timer to monitor for a navigation button double tap for the navigation contextual hint
+    func startNavigationButtonDoubleTapTimer() {
```

Comment:
I think the timer logic is correct but a bit misleading. Basically we get the information that a tap happened, and while a timer is active, don't send the action. Is that correct?

Also I'm curious where do we check that we already showed the hint? It seems here that the action will be send every-time the timer resets.

Reply:
The timer starts when one of the navigation buttons are pressed, so if a second button is pressed while the timer is running, we know we had a double tap and can present the hint.

Contextual hints are only displayed once, we have a [profile pref that is checked before presenting](https://github.com/mozilla-mobile/firefox-ios/blob/c028e7b3388af37fe7b065b409fd01d2da5a171f/firefox-ios/Client/Frontend/ContextualHint/ContextualHintEligibilityUtility.swift#L142-L146).